src/document.rs

```diff
@@ -129,172 +172,447 @@ pub struct Prescription<'a> {
     pub document: Document<'a>,
 }
 
-mod helpers {
-    use super::unicode_segmentation::UnicodeSegmentation;
-    use super::NodeType;
-
-    /// Sequenceable node types are those which can be used as a pattern in the prescription
-    /// that can match against an arbitrary sequence of nodes of the same type in a document.
-    pub fn node_type_is_sequencable(node_type: &NodeType) -> bool {
-        match *node_type {
-            NodeType::CMarkNodeParagraph
-            | NodeType::CMarkNodeHeading
-            | NodeType::CMarkNodeItem
-            | NodeType::CMarkNodeList
-            | NodeType::CMarkNodeCodeBlock => true,
-            _ => false,
-        }
+/// A circumstantial node is one that may or may not appear in the document due to its presence
+/// only being necessary if an optional child node is present.
+pub fn node_type_is_circumstantial(node: &Node) -> HowserResult<bool> {
+    let node_type = node.capabilities
+        .get
+        .as_ref()
+        .ok_or(HowserError::CapabilityError)?
+        .get_type()?;
+    match node_type {
+        NodeType::CMarkNodeList => Ok(true),
+        _ => Ok(false),
     }
+}
+
+pub fn reverse(s: &String) -> String {
```

Comment:
... what? Is this somehow not in `std`?

Reply:
`std` does not deal well with unicode character boundaries. The need for this function was removed though so it is gone.